app/talk/active-users.jsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import { sugarApiClient } from 'panoptes-client/lib/sugar';
+import { Link } from 'react-router';
+import Paginator from './lib/paginator';
+import activeUserCache from './lib/active-user-cache';
+
+export default class ActiveUsers extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      page: 1,
+      pageCount: 0,
+      perPage: 10,
+      total: 0,
+      userRecords: activeUserCache,
+      users: []
+    };
+
+    this.boundedPage = this.boundedPage.bind(this);
+    this.fetchUncachedUsers = this.fetchUncachedUsers.bind(this);
+    this.getActiveUserIds = this.getActiveUserIds.bind(this);
+    this.onPageChange = this.onPageChange.bind(this);
+    this.resetTimer = this.resetTimer.bind(this);
+    this.restartTimer = this.restartTimer.bind(this);
+    this.pageCount = this.pageCount.bind(this);
+    this.update = this.update.bind(this);
+    this.userIdsOnPage = this.userIdsOnPage.bind(this);
+    this.userLink = this.userLink.bind(this);
+  }
```

Comment:
Only event handlers need to be explicitly bound to the `this` context. I think you can delete all of these bind calls, because the constructor already binds all your instance methods for you.

Reply:
OK, you need to bind `this.onPageChange`, but should be fine to delete all the other bind calls.

Reply:
Oh nice. I had a misunderstanding of how the function binding worked.

Reply:
No problem. You have to bind handlers to the component instance because events set `this` to the element that fired the event, for historical reasons. MDN has a longer explanation. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this